test/python/pulse/test_transforms.py

```diff
@@ -401,12 +407,286 @@ def test_multiple_schedules(self):
 
 
+class TestAlignSequential(QiskitTestCase):
+    """Test sequential alignment transform."""
+
+    def test_align_sequential(self):
+        """Test sequential alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_sequential_with_barrier(self):
+        """Test sequential alignment with a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule.append(directives.RelativeBarrier(d0, d1), mutate=True)
+        schedule = schedule.insert(4, instructions.Delay(5, d1), mutate=True)
+        schedule = schedule.insert(12, instructions.Delay(7, d0), mutate=True)
+
+        schedule = transforms.remove_directives(
+            transforms.align_sequential(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+
+class TestAlignLeft(QiskitTestCase):
+    """Test left alignment transform."""
+
+    def test_align_right(self):
+        """Test left alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
+
+        schedule = transforms.align_left(schedule)
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+
```

Comment:
insert is much easier to follow to compare start times

Reply:
These occur one after the other and start and that's the information I'm trying to convey. Otherwise, the insert time is dependent on the length of the pulse that comes before which adds cognitive burden.